Submitted By: David Ciecierski <davidcie at klub.chip.pl>
Date: 2005-11-09
Initial Package Version: 2.0.0
Upstream Status: N/A
Description: This patch should enable you to build OOo2.0.0 with 
Firefox / Thunderbird that are already installed on your system. 
It's basically a modification to the configure script of xmlsec. 
Beware it has not been fully tested, feel free to modify and let 
us know if something's broken.

diff -Naur OOo_2.0.0_src_orig/libxmlsec/xmlsec1-1.2.6.patch OOo_2.0.0_src/libxmlsec/xmlsec1-1.2.6.patch
--- OOo_2.0.0_src_orig/libxmlsec/xmlsec1-1.2.6.patch	2005-08-18 09:24:35.000000000 +0200
+++ OOo_2.0.0_src/libxmlsec/xmlsec1-1.2.6.patch	2005-10-21 10:38:53.000000000 +0200
@@ -37,7 +37,11 @@
   LIBXML_LIBS=""
   LIBXML_FOUND="no"
 ***************
-*** 25682,25688 ****
+*** 25678,25688 ****
+  
+  XMLSEC_NO_NSS="1"
+  MOZILLA_MIN_VERSION="1.4"
+  NSS_MIN_VERSION="3.2"
   NSPR_MIN_VERSION="4.0"
   NSS_CFLAGS=""
   NSS_LIBS=""
@@ -45,7 +49,13 @@
   NSPR_LIBS_LIST="-lnspr4 -lplds4 -lplc4"
   NSS_CRYPTO_LIB="$PACKAGE-nss"
   NSS_FOUND="no"
---- 25682,25688 ----
+--- 25678,25690 ----
+  
+  XMLSEC_NO_NSS="1"
+  MOZILLA_MIN_VERSION="1.4"
++ FIREFOX_MIN_VERSION="1.0"
++ THUNDERBIRD_MIN_VERSION="1.0"
+  NSS_MIN_VERSION="3.2"
   NSPR_MIN_VERSION="4.0"
   NSS_CFLAGS=""
   NSS_LIBS=""
@@ -54,6 +64,129 @@
   NSS_CRYPTO_LIB="$PACKAGE-nss"
   NSS_FOUND="no"
 ***************
+*** 25766,25800 ****
+    else
+       PKG_CONFIG_MIN_VERSION=0.9.0
+       if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
+          echo "$as_me:$LINENO: checking for mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION" >&5
+! echo $ECHO_N "checking for mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION... $ECHO_C" >&6
+  
+          if $PKG_CONFIG --exists "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION" ; then
+              echo "$as_me:$LINENO: result: yes" >&5
+! echo "${ECHO_T}yes" >&6
+              succeeded=yes
+  
+              echo "$as_me:$LINENO: checking NSS_CFLAGS" >&5
+! echo $ECHO_N "checking NSS_CFLAGS... $ECHO_C" >&6
+              NSS_CFLAGS=`$PKG_CONFIG --cflags "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
+              echo "$as_me:$LINENO: result: $NSS_CFLAGS" >&5
+! echo "${ECHO_T}$NSS_CFLAGS" >&6
+  
+              echo "$as_me:$LINENO: checking NSS_LIBS" >&5
+! echo $ECHO_N "checking NSS_LIBS... $ECHO_C" >&6
+              NSS_LIBS=`$PKG_CONFIG --libs "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
+              echo "$as_me:$LINENO: result: $NSS_LIBS" >&5
+! echo "${ECHO_T}$NSS_LIBS" >&6
+!         else
+!             NSS_CFLAGS=""
+!             NSS_LIBS=""
+!             ## If we have a custom action on failure, don't print errors, but
+!             ## do set a variable so people can do so.
+!             NSS_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
+! 
+!         fi
+! 
+  
+  
+       else
+          echo "*** Your version of pkg-config is too old. You need version $PKG_CONFIG_MIN_VERSION or newer."
+--- 25768,25852 ----
+    else
+       PKG_CONFIG_MIN_VERSION=0.9.0
+       if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
++      
+          echo "$as_me:$LINENO: checking for mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION" >&5
+! 		echo $ECHO_N "checking for mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION... $ECHO_C" >&6
+  
++ 		# Check for mozilla first
+          if $PKG_CONFIG --exists "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION" ; then
+              echo "$as_me:$LINENO: result: yes" >&5
+! 			echo "${ECHO_T}yes" >&6
+              succeeded=yes
+  
+              echo "$as_me:$LINENO: checking NSS_CFLAGS" >&5
+! 			echo $ECHO_N "checking NSS_CFLAGS... $ECHO_C" >&6
+              NSS_CFLAGS=`$PKG_CONFIG --cflags "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
+              echo "$as_me:$LINENO: result: $NSS_CFLAGS" >&5
+! 			echo "${ECHO_T}$NSS_CFLAGS" >&6
+  
+              echo "$as_me:$LINENO: checking NSS_LIBS" >&5
+! 			echo $ECHO_N "checking NSS_LIBS... $ECHO_C" >&6
+              NSS_LIBS=`$PKG_CONFIG --libs "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
+              echo "$as_me:$LINENO: result: $NSS_LIBS" >&5
+! 			echo "${ECHO_T}$NSS_LIBS" >&6
+! 		fi
+  
++ 		if test $succeeded = no; then
++ 			# Check if Firefox is present and can be used instead of Mozilla
++ 			echo "$as_me:$LINENO: checking for firefox-nspr >= $FIREFOX_MIN_VERSION firefox-nss >= $FIREFOX_MIN_VERSION" >&5
++ 			echo $ECHO_N "checking for firefox-nspr >= $FIREFOX_MIN_VERSION firefox-nss >= $FIREFOX_MIN_VERSION... $ECHO_C" >&6
++ 
++ 			if $PKG_CONFIG --exists "firefox-nspr >= $FIREFOX_MIN_VERSION firefox-nss >= $FIREFOX_MIN_VERSION" ; then
++ 				echo "$as_me:$LINENO: result: yes" >&5
++ 				echo "${ECHO_T}yes" >&6
++ 				succeeded=yes
++ 				
++ 				echo "$as_me:$LINENO: checking NSS_CFLAGS" >&5
++ 				echo $ECHO_N "checking NSS_CFLAGS... $ECHO_C" >&6
++ 				NSS_CFLAGS=`$PKG_CONFIG --cflags "firefox-nspr >= $FIREFOX_MIN_VERSION firefox-nss >= $FIREFOX_MIN_VERSION"`
++ 				echo "$as_me:$LINENO: result: $NSS_CFLAGS" >&5
++ 				echo "${ECHO_T}$NSS_CFLAGS" >&6
++ 
++ 				echo "$as_me:$LINENO: checking NSS_LIBS" >&5
++ 				echo $ECHO_N "checking NSS_LIBS... $ECHO_C" >&6
++ 				NSS_LIBS=`$PKG_CONFIG --libs "firefox-nspr >= $FIREFOX_MIN_VERSION firefox-nss >= $FIREFOX_MIN_VERSION"`
++ 				echo "$as_me:$LINENO: result: $NSS_LIBS" >&5
++ 				echo "${ECHO_T}$NSS_LIBS" >&6
++ 			fi
++ 		fi
++ 		
++ 		if test $succeeded = no; then
++ 			# Check if Thunderbird is present and can be used instead of Mozilla
++ 			echo "$as_me:$LINENO: checking for thunderbird-nspr >= $THUNDERBIRD_MIN_VERSION thunderbird-nss >= $THUNDERBIRD_MIN_VERSION" >&5
++ 			echo $ECHO_N "checking for thunderbird-nspr >= $THUNDERBIRD_MIN_VERSION thunderbird-nss >= $THUNDERBIRD_MIN_VERSION... $ECHO_C" >&6
++ 
++ 			if $PKG_CONFIG --exists "thunderbird-nspr >= $THUNDERBIRD_MIN_VERSION thunderbird-nss >= $THUNDERBIRD_MIN_VERSION" ; then
++ 				echo "$as_me:$LINENO: result: yes" >&5
++ 				echo "${ECHO_T}yes" >&6
++ 				succeeded=yes
++ 				
++ 				echo "$as_me:$LINENO: checking NSS_CFLAGS" >&5
++ 				echo $ECHO_N "checking NSS_CFLAGS... $ECHO_C" >&6
++ 				NSS_CFLAGS=`$PKG_CONFIG --cflags "thunderbird-nspr >= $THUNDERBIRD_MIN_VERSION thunderbird-nss >= $THUNDERBIRD_MIN_VERSION"`
++ 				echo "$as_me:$LINENO: result: $NSS_CFLAGS" >&5
++ 				echo "${ECHO_T}$NSS_CFLAGS" >&6
++ 
++ 				echo "$as_me:$LINENO: checking NSS_LIBS" >&5
++ 				echo $ECHO_N "checking NSS_LIBS... $ECHO_C" >&6
++ 				NSS_LIBS=`$PKG_CONFIG --libs "thunderbird-nspr >= $THUNDERBIRD_MIN_VERSION thunderbird-nss >= $THUNDERBIRD_MIN_VERSION"`
++ 				echo "$as_me:$LINENO: result: $NSS_LIBS" >&5
++ 				echo "${ECHO_T}$NSS_LIBS" >&6
++ 			fi
++ 		fi
++ 	
++ 		if test $succeeded = no; then
++ 			# None of Mozilla, Firefox or Thunderbird were found
++ 			NSS_CFLAGS=""
++ 			NSS_LIBS=""
++ 			## If we have a custom action on failure, don't print errors, but
++ 			## do set a variable so people can do so.
++ 			NSS_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "mozilla-nspr >= $MOZILLA_MIN_VERSION mozilla-nss >= $MOZILLA_MIN_VERSION"`
++ 		fi
+  
+       else
+          echo "*** Your version of pkg-config is too old. You need version $PKG_CONFIG_MIN_VERSION or newer."
+***************
 *** 25817,25824 ****
           ac_mozilla_name=mozilla-$MOZILLA_MIN_VERSION
       fi
@@ -63,7 +196,7 @@
   
       echo "$as_me:$LINENO: checking for nspr libraries >= $NSPR_MIN_VERSION" >&5
   echo $ECHO_N "checking for nspr libraries >= $NSPR_MIN_VERSION... $ECHO_C" >&6
---- 25817,25824 ----
+--- 25869,25876 ----
           ac_mozilla_name=mozilla-$MOZILLA_MIN_VERSION
       fi
   
@@ -81,7 +214,7 @@
   		        	if test "z$dir" = "z/usr/lib" ; then
               	    NSPR_LIBS="$NSPR_LIBS_LIST"
       		else
---- 25853,25859 ----
+--- 25905,25911 ----
   	done
   
   	for dir in $ac_nss_lib_dir ; do
@@ -98,7 +231,7 @@
           	    		if test "z$dir" = "z/usr/lib" ; then
           	    NSS_LIBS="$NSS_LIBS_LIST"
                   else
---- 25939,25945 ----
+--- 25991,25997 ----
           done
   
           for dir in $ac_nss_lib_dir ; do
